### DISCUSSION TITLE
Custom Dataset Training Directory Structure

### DISCUSSION THREAD

**Discussion by nicholasjprimianomd** [mobaidoctor/med-ddpm#24] [Q&A]
What is the exact directory structure needed for training on a custom dataset?

**Comment by mobaidoctor** [comment]
@nicholasjprimianomd Thank you for your inquiry. The exact directory structure needed for training on a custom dataset depends on the modalities available in your dataset. You can use the template folder structure provided in the 'dataset' directory of our repository. Here are specific examples:
- Single Modality Synthesis from a Segmentation Mask: Use the dataset/whole_head structure, which includes image and mask subfolders. Place your images and masks in these respective subfolders.
- Multi-Modality Training (similar to the BRATS dataset): For training with four modalities plus a mask, like the BRATS dataset, use the structure found under dataset/brats2021. This includes subfolders for flair, t1, t1ce, t2, and seg. Place your images in the corresponding subfolders.

If your dataset includes different modalities or requires a different structure, you will need to adjust the directory structure accordingly. Additionally, you should update the train.py and dataset.py files to reflect these changes.
